docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,63 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, namespace) to individual log entries, and sending the logs to the Elasticsearch cluster. The pod metadata is sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  This results in logs not being shipped to the aggregator and thus being unavailable.
+
+##Proposal - Short Term (3.4)
+1. Leave the value as-is and note that it must be adjusted to account for # of nodes plus some headroom (swag @ 10%?).
+1. Increase the default to our current maximum supported node count, plus some headroom.
+1. Increasing maxRequestsInflight upstream
+
+##Proposal - Long Term
+###Option 1 - Admission Controller
+1. Create a simple admission controller which simply embeds the necessary meta-data as env-variables into the pods on submission.  In this way every pod that is run on the system will have its data locally.
+1. Mount the docker socket into the Fluentd pod such that it can query containers env-variables.
```

Comment:
I like that idea more.  And perhaps this already works (although returning a list and not a specific pod)

```
$ curl -k https://0.0.0.0:10250/pods
{"kind":"PodList","apiVersion":"v1","metadata":{},"items":[{"metadata":{"name":"nginx","namespace":"default","selfLink":"/api/v1/namespaces/default/pods/nginx","uid":"61ca8777-9237-11e6-bfb5-54ee752009cb","resourceVersion":"63","creationTimestamp":"2016-10-14T17:55:24Z","labels":{"name":"nginx"},"annotations":{"kubernetes.io/config.seen":"2016-10-14T13:55:31.298735728-04:00","kubernetes.io/config.source":"api"}},"spec":{"volumes":[{"name":"default-token-ksyfd","secret":{"secretName":"default-token-ksyfd","defaultMode":420}}],"containers":[{"name":"nginx","image":"nginx","ports":[{"containerPort":80,"protocol":"TCP"}],"resources":{},"volumeMounts":[{"name":"default-token-ksyfd","readOnly":true,"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount"}],"terminationMessagePath":"/dev/termination-log","imagePullPolicy":"Always"}],"restartPolicy":"Always","terminationGracePeriodSeconds":30,"dnsPolicy":"ClusterFirst","serviceAccountName":"default","serviceAccount":"default","nodeName":"127.0.0.1","securityContext":{}},"status":{"phase":"Running","conditions":[{"type":"Initialized","status":"True","lastProbeTime":null,"lastTransitionTime":"2016-10-14T17:55:31Z"},{"type":"Ready","status":"True","lastProbeTime":null,"lastTransitionTime":"2016-10-14T17:55:34Z"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2016-10-14T17:55:31Z"}],"hostIP":"127.0.0.1","podIP":"172.17.0.2","startTime":"2016-10-14T17:55:31Z","containerStatuses":[{"name":"nginx","state":{"running":{"startedAt":"2016-10-14T17:55:33Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"nginx","imageID":"docker-pullable://nginx@sha256:e40499ca855c9edfb212e1c3ee1a6ba8b2d873a294d897b4840d49f94d20487c","containerID":"docker://39e9b03e617c7d136ec132703f6bba673b12c8f4a11acd3bbf2edbab79eb6541"}]}}]}
```

If a handler was installed by default that was equivalent of the `GetPodByName` call that the `HostInterface` already supports it would definitely be handy.  This seems like a small change that would allow us to leverage anything the Kubelet uses for getting to the information.